core/trino-main/src/main/java/io/trino/sql/gen/RowExpressionCompiler.java

```diff
@@ -135,6 +135,9 @@ public BytecodeNode visitSpecialForm(SpecialForm specialForm, Context context)
                 case ROW_CONSTRUCTOR:
                     generator = new RowConstructorCodeGenerator(specialForm);
                     break;
+                case ARRAY_CONSTRUCTOR:
```

Comment:
nit: this switch could be modernized to enhanced one

Reply:
Yeah, I’ll do it separately.